rest/replication_api_test.go

```diff
@@ -405,3 +409,353 @@ func TestReplicationsFromConfig(t *testing.T) {
+	_ = remoteRT.putDoc(docDEF1, `{"source":"remoteRT","channels":["DEF"]}`)
+
+	// Create pull replications, verify running
+	activeRT.createReplication("rep_ABC", remoteURLString, db.ActiveReplicatorTypePull, []string{"ABC"})
+	activeRT.assertReplicationState("rep_ABC", db.ReplicationStateRunning)
+	activeRT.createReplication("rep_DEF", remoteURLString, db.ActiveReplicatorTypePull, []string{"DEF"})
+	activeRT.assertReplicationState("rep_DEF", db.ReplicationStateRunning)
+
+	// wait for documents originally written to remoteRT to arrive at activeRT
+	changesResults := activeRT.RequireWaitChanges(2, "0")
+	changesResults.requireDocIDs(t, []string{docABC1, docDEF1})
+
+	// Validate doc contents
+	docABC1Body := activeRT.getDoc(docABC1)
+	assert.Equal(t, "remoteRT", docABC1Body["source"])
+	docDEF1Body := activeRT.getDoc(docDEF1)
+	assert.Equal(t, "remoteRT", docDEF1Body["source"])
+
+	// Add another node to the active cluster
+	activeRT2 := addActiveRT(t, activeRT.TestBucket)
+	defer activeRT2.Close()
+
+	// Wait for replication to be rebalanced to activeRT2
+	activeRT.waitForAssignedReplications(1)
+	activeRT2.waitForAssignedReplications(1)
+
+	// Create additional docs on remoteRT
+	docABC2 := t.Name() + "ABC2"
+	_ = remoteRT.putDoc(docABC2, `{"source":"remoteRT","channels":["ABC"]}`)
+	docDEF2 := t.Name() + "DEF2"
+	_ = remoteRT.putDoc(docDEF2, `{"source":"remoteRT","channels":["DEF"]}`)
+
+	// wait for new documents to arrive at activeRT
+	changesResults = activeRT.RequireWaitChanges(2, changesResults.Last_Seq.(string))
+	changesResults.requireDocIDs(t, []string{docABC2, docDEF2})
+
+	// Validate doc contents
+	docABC2Body := activeRT.getDoc(docABC2)
+	assert.Equal(t, "remoteRT", docABC2Body["source"])
+	docDEF2Body := activeRT.getDoc(docDEF2)
+	assert.Equal(t, "remoteRT", docDEF2Body["source"])
+	docABC2Body2 := activeRT2.getDoc(docABC2)
+	assert.Equal(t, "remoteRT", docABC2Body2["source"])
+	docDEF2Body2 := activeRT2.getDoc(docDEF2)
+	assert.Equal(t, "remoteRT", docDEF2Body2["source"])
+}
+
+// TestReplicationRebalancePush
+//   - Starts 2 RestTesters, one active, and one passive.
+//   - Creates documents on rt1 in two channels
+//   - Creates two continuous pull replications on rt1 via the REST API
+//   - adds another active node
+//   - Creates more documents, validates they are replicated
+func TestReplicationRebalancePush(t *testing.T) {
+
+	if base.GTestBucketPool.NumUsableBuckets() < 2 {
+		t.Skipf("test requires at least 2 usable test buckets")
+	}
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate, base.KeyHTTP, base.KeyHTTPResp, base.KeySync, base.KeySyncMsg)()
+
+	activeRT, remoteRT, remoteURLString, teardown := setupSGRPeers(t)
+	defer teardown()
+
+	// Create docs on active
+	docABC1 := t.Name() + "ABC1"
+	docDEF1 := t.Name() + "DEF1"
+	_ = remoteRT.putDoc(docABC1, `{"source":"activeRT","channels":["ABC"]}`)
+	_ = remoteRT.putDoc(docDEF1, `{"source":"activeRT","channels":["DEF"]}`)
```

Comment:
These need to be created on `activeRT`, but when I change it, I'm seeing weird behaviour where `docABC2` isn't sent without a sleep on line 615

```suggestion
	_ = activeRT.putDoc(docABC1, `{"source":"activeRT","channels":["ABC"]}`)
	_ = activeRT.putDoc(docDEF1, `{"source":"activeRT","channels":["DEF"]}`)
```

Reply:
Thanks for catching - I'll take a look.

Reply:
Made that fix, and added handling to avoid the race in CBG-1000 until it's fixed.